File: python/examples/example_agent_invocation.py
#!/usr/bin/env python3
"""
Example script demonstrating how to use ClaudeAgents to build a complete software solution.

This example shows how to:
1. Set up and configure the manager agent
2. Execute a complex development task using all agent types
3. Process and display the results
"""

import os
import asyncio
import json
from loguru import logger

from claude_agents.utils.logging_setup import setup_logging
from claude_agents.utils.memory_manager import MemoryManager
from claude_agents.agents.manager_agent import ManagerAgent


async def run_example():
    """Run a complete example using all agent types."""
    # Setup logging
    setup_logging(log_file="logs/example.log", log_level="INFO")
    
    # Initialize memory manager with persistence
    memory_manager = MemoryManager(persist_to_disk=True, storage_dir="./memory")
    
    # Create the manager agent
    manager = ManagerAgent(name="Manager", memory_manager=memory_manager)
    
    # Example prompt that will require all agent functions
    prompt = """
    Create a weather dashboard application with the following features:
    
    1. A responsive web interface that displays current weather and 5-day forecast
    2. Backend API that fetches data from a public weather API
    3. Ability to search for weather by city name or zip code
    4. Show temperature, humidity, wind speed, and weather conditions
    5. Store user's recent searches
    6. Display weather data using charts and icons
    7. Automatically detect user's location on initial load
    
    The solution should be easy to deploy and use Docker for containerization.
    Make it accessible for developers to extend with new features.
    """
    
    print("\n" + "=" * 60)
    print(" ClaudeAgents - Multi-agent Collaborative Development Example ")
    print("=" * 60 + "\n")
    
    print(f"🚀 Starting development process with prompt:\n{prompt}\n")
    print("📋 This example will demonstrate the complete collaborative workflow using all agent types:\n")
    print("1. Manager Agent - Overall coordination")
    print("2. Architecture Designer - System architecture")
    print("3. Stack Builder - Technology stack setup")
    print("4. Library Researcher - Finding optimal libraries")
    print("5. Software Planner - Development planning")
    print("6. Software Programmer - Code implementation")
    print("7. Exception Debugger - Testing and debugging")
    print("8. Dependency Analyzer - Dependency management\n")
    
    # Execute the manager agent with the prompt
    try:
        print("⏳ Starting multi-agent process - this may take several minutes...\n")
        result = await manager.execute(prompt)
        
        if result["success"]:
            print("\n✅ Process completed successfully!\n")
            
            # Access detailed results
            final_result = result.get("result", {})
            
            # Print summary
            print("📑 Solution Summary:")
            print(f"  {final_result.get('summary', 'No summary available')[:500]}...\n")
            
            # Print architecture overview
            print("🏗️ Architecture Overview:")
            architecture = final_result.get("architecture", {})
            print(f"  {architecture.get('summary', 'No architecture details available')[:300]}...\n")
            
            # Print technology stack
            libraries = final_result.get("libraries", {})
            if isinstance(libraries, dict) and "libraries" in libraries:
                print("📚 Selected Libraries:")
                for lib in libraries["libraries"]:
                    print(f"  • {lib.get('name', 'Unknown')}: {lib.get('description', 'No description')[:100]}")
            
            # Print file summary
            files = final_result.get("files", {})
            if files:
                print("\n💻 Generated Files:")
                for filename, content in files.items():
                    print(f"  • {filename}")
            
            print("\n📁 Full solution details saved to memory directory.")
            
        else:
            print(f"\n❌ Process failed: {result.get('error', 'Unknown error')}")
    
    except Exception as e:
        logger.exception(f"Error in example: {str(e)}")
        print(f"\n❌ Error: {str(e)}")


# Run the example
if __name__ == "__main__":
    asyncio.run(run_example())

File: javascript/examples/example-agent-invocation.js
#!/usr/bin/env node
/**
 * Example script demonstrating how to use ClaudeAgents to build a complete software solution.
 *
 * This example shows how to:
 * 1. Set up and configure the manager agent
 * 2. Execute a complex development task using all agent types
 * 3. Process and display the results
 */

const { setupLogger } = require('../src/utils/LoggerSetup');
const { MemoryManager } = require('../src/utils/MemoryManager');
const { ManagerAgent } = require('../src/agents/ManagerAgent');

async function runExample() {
  // Setup logging
  const logger = setupLogger({
    logLevel: 'info',
    logToFile: true,
    logFileDir: 'logs',
  });

  // Initialize memory manager with persistence
  const memoryManager = new MemoryManager(true, './memory');

  // Create the manager agent
  const manager = new ManagerAgent('Manager', memoryManager);

  // Example prompt that will require all agent functions
  const prompt = `
    Create a weather dashboard application with the following features:
    
    1. A responsive web interface that displays current weather and 5-day forecast
    2. Backend API that fetches data from a public weather API
    3. Ability to search for weather by city name or zip code
    4. Show temperature, humidity, wind speed, and weather conditions
    5. Store user's recent searches
    6. Display weather data using charts and icons
    7. Automatically detect user's location on initial load
    
    The solution should be easy to deploy and use Docker for containerization.
    Make it accessible for developers to extend with new features.
  `;

  console.log('\n' + '='.repeat(60));
  console.log(' ClaudeAgents - Multi-agent Collaborative Development Example ');
  console.log('='.repeat(60) + '\n');

  console.log(`🚀 Starting development process with prompt:\n${prompt}\n`);
  console.log('📋 This example will demonstrate the complete collaborative workflow using all agent types:\n');
  console.log('1. Manager Agent - Overall coordination');
  console.log('2. Architecture Designer - System architecture');
  console.log('3. Stack Builder - Technology stack setup');
  console.log('4. Library Researcher - Finding optimal libraries');
  console.log('5. Software Planner - Development planning');
  console.log('6. Software Programmer - Code implementation');
  console.log('7. Exception Debugger - Testing and debugging');
  console.log('8. Dependency Analyzer - Dependency management\n');

  // Execute the manager agent with the prompt
  try {
    console.log('⏳ Starting multi-agent process - this may take several minutes...\n');
    const result = await manager.execute(prompt);

    if (result.success) {
      console.log('\n✅ Process completed successfully!\n');

      // Access detailed results
      const finalResult = result.result || {};

      // Print summary
      console.log('📑 Solution Summary:');
      console.log(`  ${(finalResult.summary || 'No summary available').substring(0, 500)}...\n`);

      // Print architecture overview
      console.log('🏗️ Architecture Overview:');
      const architecture = finalResult.architecture || {};
      console.log(`  ${(architecture.summary || 'No architecture details available').substring(0, 300)}...\n`);

      // Print technology stack
      const libraries = finalResult.libraries || {};
      if (libraries.libraries && Array.isArray(libraries.libraries)) {
        console.log('📚 Selected Libraries:');
        for (const lib of libraries.libraries) {
          console.log(`  • ${lib.name || 'Unknown'}: ${(lib.description || 'No description').substring(0, 100)}`);
        }
      }

      // Print file summary
      const files = finalResult.files || {};
      if (Object.keys(files).length > 0) {
        console.log('\n💻 Generated Files:');
        for (const filename of Object.keys(files)) {
          console.log(`  • ${filename}`);
        }
      }

      console.log('\n📁 Full solution details saved to memory directory.');
    } else {
      console.log(`\n❌ Process failed: ${result.error || 'Unknown error'}`);
    }
  } catch (error) {
    logger.error(`Error in example: ${error.message}`);
    console.log(`\n❌ Error: ${error.message}`);
  }
}

// Run the example
runExample().catch(error => {
  console.error('Unhandled error in example:', error);
});